src/main/java/org/gaul/s3proxy/Main.java

```diff
@@ -386,6 +388,20 @@ private static BlobStore createBlobStore(Properties properties,
             builder = builder.endpoint(endpoint);
         }
 
+        if ((identity.isEmpty() || credential.isEmpty()) && provider.equals("aws-s3")) {
+            DefaultAWSCredentialsProviderChain creds = DefaultAWSCredentialsProviderChain.getInstance();
+            Supplier<Credentials> credentialsSupplier = new Supplier<Credentials>() {
+                @Override
+                public Credentials get() {
+                    return new Credentials(creds.getCredentials().getAWSAccessKeyId(),
+                            creds.getCredentials().getAWSSecretKey());
+                }
+            };
```

Comment:
Could you use the Apache jclouds STS supplier for consistency:

https://stackoverflow.com/questions/23520216/using-aws-s3-via-jclouds-how-to-assume-role